src/chainstate/stacks/block.rs

```diff
@@ -1044,12 +1044,12 @@ mod test {
     #[test]
     fn codec_stacks_microblock() { 
         // make a block with each and every kind of transaction
-        let mut all_txs = codec_all_transactions(&TransactionVersion::Testnet, 0x80000000, &TransactionAnchorMode::OffChainOnly, &TransactionPostConditionMode::Allow);
+        let all_txs = codec_all_transactions(&TransactionVersion::Testnet, 0x80000000, &TransactionAnchorMode::OffChainOnly, &TransactionPostConditionMode::Allow);
 
         // remove all coinbases
         let mut txs_anchored = vec![];
 
-        for tx in all_txs.drain(..) {
+        for tx in all_txs.iter() {
```

Comment:
Why switch to a non-draining iterator?

Reply:
`all_txs.len()` needs to be non-zero -- see line 1063.

Reply:
Gotcha -- I also see now that this in a test anyways.